src/cofi_espresso/espresso_problem.py

```diff
@@ -1,37 +1,238 @@
+                orig_init_subclass(**kwargs)
+            for name in names:
+                if getattr(cls, name, NotImplemented) is NotImplemented:
+                    raise NotImplementedError(
+                        f"{name} is required as a class attribute but you haven't defined it"
+                    )
+        cls.__init_subclass__ = classmethod(new_init_subclass)
+        return cls
+
+    return lambda cls: _func(cls, *names)
+
+
+@abstract_class_attributes(
+    "problem_title",
+    "problem_short_description",
+    "author_names",
+    "contact_name",
+    "contact_email",
+    "citations",
+    "linked_sites",
+)
 class EspressoProblem(metaclass=ABCMeta):
     """Base class for all Espresso problems
 
     All Espresso problems shoud be a subclass of this class.
     """
 
-    def __init__(self, example_number=0):
+    def __init__(self, example_number=1):
         self.example_number = example_number
         self.params = dict()
+
+    @property
+    def description(self):
+        """
+        Returns (desc,) with
+        
+        desc - str -- A string containing a brief (1-3 sentence) 
+            description of the example.
+        """
+        raise NotImplementedError
+
+    @property
+    @abstractmethod
+    def model_size(self):
+        """
+        Returns (M,) with
+
+        M - integer -- The number of model parameters (i.e. the
+            dimension of a model vector).
+        """
+        raise NotImplementedError
     
+    @property
     @abstractmethod
-    def suggested_model(self):
+    def data_size(self):
+        """
+        Returns (N,) with
+
+        N - integer -- The number of data points (i.e. the
+            dimension of a data vector).
+        """
+        raise NotImplementedError
+
+    @property
+    @abstractmethod
+    def good_model(self):
+        """
+        Returns (m,) with
+        
+        m - np.array, shape(model_size) -- A model vector that the 
+            contributor regards as being a 'correct' or 'sensible' 
+            explanation of the dataset. (In some problems it may 
+            be the case that there are many 'equally good' models.
+            The contributor should select just one of these.)
+        """
+        raise NotImplementedError
+
+    @property
+    @abstractmethod
+    def starting_model(self):
+        """
+        Returns (m,) with
+
+        m - np.array, shape(model_size) -- A model vector, possibly
+            just np.zeros(model_size), representing a typical 
+            starting point or 'null model' for an inversion.
+        """
         raise NotImplementedError
 
+    @property
     @abstractmethod
     def data(self):
+        """
+        Returns (d,) with
+        
+        d - np.array, shape(data_size) -- A data vector in the same
+            format as output by forward().
+        """
         raise NotImplementedError
-    
+
+    @property
+    @abstractmethod
+    def covariance_matrix(self):
```

Comment:
Is it always necessary? Does it increase the barrier to contributing?

Reply:
I guess we could make it optional...